### PR TITLE
Vasili Makouchyk - 1

### PR DESCRIPTION
**Pull request by weesom**
## Name
Vasili Makouchyk

## Homework
1

### CODE REVIEW

File: Vasili Makouchyk/1/database.rb [hunk 1]
```diff
@@ -0,0 +1,39 @@
+require 'nokogiri'
+require 'mechanize'
+require 'redis'
+require 'open-uri'
+
+URL = 'http://rapstyle.su/quotes.php'.freeze
```
Comment (RGBD) [inline Vasili Makouchyk/1/database.rb]: move inside database class

File: Vasili Makouchyk/1/database.rb [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  end
+
+  def download_page
+    agent = Mechanize.new
+    page = agent.get(URL)
+    File.write(FILE, page.body)
+  end
+
+  def scrape_quotes
+    page = Nokogiri::HTML(File.open(FILE)) # File.read('citaty.html'))
```
Comment (RGBD) [inline Vasili Makouchyk/1/database.rb]: remove comment

File: Vasili Makouchyk/1/telegram_bot.rb [hunk 3]
```diff
@@ -0,0 +1,61 @@
+require 'telegram/bot'
+require_relative 'database.rb'
+
+TOKEN = File.read('token.txt').intern
```
Comment (RGBD) [inline Vasili Makouchyk/1/telegram_bot.rb]: move token inside bot class

File: Vasili Makouchyk/1/database.rb [hunk 4]
```diff
@@ -0,0 +1,39 @@
+class DataBase
+  require 'nokogiri'
```
Comment (RGBD) [inline Vasili Makouchyk/1/database.rb]: put requires at the top of the file

File: Vasili Makouchyk/1/database.rb [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    File.write(FILE, page.body)
+  end
+
+  def scrape_quotes
+    page = Nokogiri::HTML(File.open(FILE))
+    @quotes = page.css('.post-entry p')
+  end
+
+  def set_db
+    abort("Quotes hadn't been scraped") if @quotes.empty?
```
Comment (RGBD) [inline Vasili Makouchyk/1/database.rb]: You almost never need abort (at least in ruby).
https://en.wikipedia.org/wiki/Abort_(computing)
Why not use raise and let anyone calling your code handle the situation.
Also empty quotes doesn't seem like something that should be taken care of internally.
tl;dr use raise or remove the line

File: Vasili Makouchyk/1/telegram_bot.rb [hunk 6]
```diff
@@ -0,0 +1,58 @@
+class TelegramBot
+  require 'telegram/bot'
```
Comment (RGBD) [inline Vasili Makouchyk/1/telegram_bot.rb]: put requires at the top of the file

File: Vasili Makouchyk/1/telegram_bot.rb [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+  require_relative 'database.rb'
+
+  TOKEN = File.read('token.txt').intern
```
Comment (RGBD) [inline Vasili Makouchyk/1/telegram_bot.rb]: requiring your file shouldn't raise an error.
better move file read into initializer

File: Vasili Makouchyk/1/telegram_bot.rb [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+
+  def initialize
+    @db = DataBase.new
+    @bot = Telegram::Bot::Client.new(TOKEN)
+    run
```
Comment (RGBD) [inline Vasili Makouchyk/1/telegram_bot.rb]: don't do any heavy work in initializer, call run later explicitly:
Bot.new.run

File: Vasili Makouchyk/1/database.rb [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+  require 'redis'
+  require 'open-uri'
+
+  URL = 'http://rapstyle.su/quotes.php'.freeze
+  FILE = 'citaty.html'.freeze
+
+  def initialize
+    @database = Redis.new
+    @quotes = []
+    download_page
```
Comment (RGBD) [inline Vasili Makouchyk/1/database.rb]: You're breaking single responsibility principle.
This class takes care of both presistence and scraping.
Ideal option is to make two separate classes.
It would be enough to move scraping into separate method:
```
def initialize
  @database = Redis.new
end

def load_quotes
  ....find quotes, persist....
  self
end
```
So in bot class you could write something like:
```@database = Database.new.load_quotes```

**Comment by weesom** [comment]
Is it ok having problems with config file?

File: Vasili Makouchyk/1/telegram_bot.rb [hunk 10]
```diff
@@ -0,0 +1,58 @@
+require 'telegram/bot'
+require_relative 'database.rb'
+
+class TelegramBot
+  def initialize
+    @token = File.read('token.txt').intern
```
Comment (RGBD) [inline Vasili Makouchyk/1/telegram_bot.rb]: use chomp to remove line ending
why did you need #intern in the first place?

**Comment by RGBD** [comment]
Aw3some)
